qemu/tests/slof_next_entry.py

```diff
@@ -0,0 +1,91 @@
+import logging
+
+from virttest import error_context
+
+from provider import slof
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Verify that next-entry will get unset after reboot.
+
+    Step:
+     1) Check if any error info in output of SLOF during booting.
+     2) Ensure the guest has at least two kernel versions.
+     3) Set a boot next entry and check it.
+     4) Reboot guest, check the kernel version and value of next_entry.
+     5) Reboot guest again, continue to check the kernel version.
+
+    :param test: Qemu test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+
+    def get_kernels_info():
+        """ Get detailed information about each kernel version in the guest. """
+        kernels_info = {}
+        for kernel in kernel_list:
+            grubby_info = session.cmd_output("grubby --info=%s" % kernel,
+                                             print_func=logging.info)
+            entry_dict = dict((item.replace('"', '').split("=", 1)
+                               for item in grubby_info.splitlines()))
+            kernels_info[int(entry_dict.pop("index"))] = entry_dict
+        return kernels_info
+
+    def check_kernel_version(k_index):
+        """ Check whether the kernel version matches the kernel index. """
+        current_kernel = session.cmd_output("uname -r").strip()
+        if guest_kernels[k_index]["kernel"].split("-", 1)[1] != current_kernel:
+            logging.info("The current kernel version is: %s", current_kernel)
+            test.fail("The current kernel version is different from expected")
+        logging.info("The kernel version matches the kernel index")
+
+    get_kernel_list_cmd = params["get_kernel_list_cmd"]
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+
+    error_context.base_context("Check the output of SLOF.", logging.info)
+    content = slof.wait_for_loaded(vm, test)[0]
+    slof.check_error(test, content)
+    session = vm.wait_for_login()
+
+    logging.info("Ensure the guest has at least two kernel versions")
+    kernel_list = session.cmd_output(get_kernel_list_cmd).splitlines()
+    if len(kernel_list) < 2:
+        test.cancel("This test requires at least two kernel versions in the "
+                    "guest")
+    if session.cmd_output("grubby --default-index").strip() != "0":
```

Comment:
I just feel that encapsulating grubby will be more clear, like the following:

```
    def _run_grubby_get_output(arguments):
        return session.cmd_output("grubby %s" % arguments)
```

Reply:
Not plan to use a function to handle only one line.

Reply:
It's ok for me, I see grubby cmd is called 3 times.